auth/src/ios/common_ios.h

```diff
@@ -91,8 +91,8 @@ static inline FIRAuth *_Nonnull AuthImpl(AuthData *_Nonnull auth_data) {
 
 /// Convert from the void* credential implementation pointer into the Obj-C
 /// FIRAuthCredential pointer.
-static inline FIRAuthCredential *_Nonnull CredentialFromImpl(void *_Nonnull impl) {
-  return static_cast<FIRAuthCredentialPointer *>(impl)->get();
+static inline FIRAuthCredential *_Nullable CredentialFromImpl(void *_Nullable impl) {
+  return impl ? static_cast<FIRAuthCredentialPointer *>(impl)->get() : nil;
```

Comment:
Also need to change the function below this one,

Reply:
Also, many iOS objective-c API expect nonnull.
https://firebase.google.com/docs/reference/ios/firebaseauth/api/reference/Classes/FIRAuth
https://github.com/firebase/firebase-cpp-sdk/blob/main/auth/src/ios/auth_ios.mm#L460

Reply:
Done